### ISSUE TITLE
figwheel-main / repl 0.2.13 & ClojureScript 1.10.844 logging initialization failure

### ISSUE THREAD

**Issue by ak-coram** [bhauman/figwheel-main#289]
I wasn't sure if I should file this issue against the `figwheel-repl` repository instead, so please let me know wether or not I should move this discussion there. Thanks!

The call to `figwheel.repl.loggin/debug` in `figwheel.repl/init-log-level!` leads to the following assertion error during REPL initialization:

```
message: "Failure: Root logger has no level set."
​
messagePattern: "Failure: Root logger has no level set."
​
reportErrorToServer: true
​
DebugError@http://localhost:3000/cljs/out/goog/debug/error.js:13:20
goog.asserts.AssertionError@http://localhost:3000/cljs/out/goog/asserts/asserts.js:64:20
goog.asserts.fail@http://localhost:3000/cljs/out/goog/asserts/asserts.js:243:32
getEffectiveLevel@http://localhost:3000/cljs/out/goog/log/log.js:198:18
getEffectiveLevel@http://localhost:3000/cljs/out/goog/log/log.js:195:28
goog.log.getEffectiveLevel@http://localhost:3000/cljs/out/goog/log/log.js:298:24
goog.log.isLoggable@http://localhost:3000/cljs/out/goog/log/log.js:304:36
goog.log.log@http://localhost:3000/cljs/out/goog/log/log.js:328:46
figwheel$repl$logging$debug@http://localhost:3000/cljs/out/figwheel/repl/logging.js:39:21
figwheel$repl$logging$set_log_level@http://localhost:3000/cljs/out/figwheel/repl/logging.js:72:29
figwheel$repl$init_log_level_BANG_@http://localhost:3000/cljs/out/figwheel/repl.js:1311:37
figwheel$repl$connect_STAR_@http://localhost:3000/cljs/out/figwheel/repl.js:1333:36
figwheel.repl.connect.cljs$core$IFn$_invoke$arity$variadic/figwheel.repl.connected<@http://localhost:3000/cljs/out/figwheel/repl.js:1383:29
figwheel.repl.connect.cljs$core$IFn$_invoke$arity$variadic@http://localhost:3000/cljs/out/figwheel/repl.js:1386:3
figwheel$repl$connect@http://localhost:3000/cljs/out/figwheel/repl.js:1372:30
@http://localhost:3000/cljs/out/figwheel/repl/preload.js:7:30
EventListener.handleEvent*@http://localhost:3000/cljs/out/figwheel/repl/preload.js:6:13
```

I think this happens because `figwheel.repl/connect` tries to set up an initial log level for the figwheel loggers, but before doing so `figwheel.repl.logging/set-log-level` tries to use these loggers itself to report about the log level change (leading to the error above).

A quick workaround is to disable assertion errors (`goog.asserts`) altogether via `:closure-defines` in your figwheel build file (e.g. `dev.cljs.edn`):

```clojure
{...
 :closure-defines {goog.asserts/ENABLE_ASSERTS false}}
```

**Comment by ak-coram** [comment]
I've looked a bit more into this and can't quite figure out how I end up with no level for the root logger in the first place. According to the closure-library's sources it's always set to `goog.log.Level.CONFIG` by default:

https://github.com/google/closure-library/blob/5ee9c7f1ee0090017f813b9e6516d4fa0fc2d0fd/closure/goog/log/log.js#L727

Yet `goog.log.getLevel(goog.log.getRootLogger())` evaluates to `null` for me. Finally I was able to set the root logger level before REPL initialization via `:preloads` in figwheel, so if this issue isn't affecting anyone else I think it can be safely closed. Nonetheless this behavior seems to be related to the dependency updates mentioned in the issue title (since this version combination is triggering it for me).
